### PR TITLE
Fixes #4099: Localize currency according to selected currency

### LINKED ISSUE #4099: Localize currency according to selected currency
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a user, I want to see currency in the selected format so that it looks correct and readable.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Improvement

*Summary*: Users can now set their desired currency in https://github.com/status-im/status-react/issues/4008 but Status doesn't show the selected currency according to localized display conventions.

#### Expected behavior
[comment]: # (Describe what you expected to happen.)
The fiat currency should be displayed as per the selected currency conventions. Eg
USD: $7.89 ($ symbol, 2 decimal places, period as decimal separator)
JPY:   ‎¥83 ($ symbol, no decimal places)

#### Actual behavior
[comment]: # (Describe what actually happened.)

1.  USD isn't shown consistently with 2 decimal places. Eg Status inconsistently shows 15 (no decimal places), 15.1 (1 decimal place) or 15.13 (2 decimal places).
2. Currency without decimal places such as JPY are incorrectly shown with decimal places like 150.05
3. Wallet "total value" doesn't show the currency symbol, but the fiat value of assets do. We should be consistent, and also show the currency symbol in the correct position.

### Additional Information

The solution should not introduce additional dependencies.

### PR DESCRIPTION
**Pull request by pacamara**
Fixes #4099: Localize currency according to selected currency

### Summary:

Display currency with correct decimal format and symbol.

### Steps to test:

* Select different currencies in Profile->Main Currency
* For each selected currency, on the wallet screen, Total Value and each asset's fiat value should have correct symbol and decimal format.
* Include some very rare currencies, currencies with different decimal formats, and currencies with different positioning (before/after amount) of the symbol.

status: ready

### CODE REVIEW

**Review by goranjovic — APPROVED** [review]

File: src/status_im/ui/screens/wallet/views.cljs [hunk 1]
```diff
@@ -28,7 +28,7 @@
     [react/view {:style styles/total-balance}
      [react/text {:style               styles/total-balance-value
                   :accessibility-label :total-amount-value-text}
-      value]
+      (i18n/format-currency (:code currency) value)]
```
Comment (dmitryn) [inline src/status_im/ui/screens/wallet/views.cljs]: I'd suggest keeping views as simple as possible and move computation to subscription https://github.com/Day8/re-frame/blob/master/docs/SubscriptionsCleanup.md

Comment (pacamara) [reply src/status_im/ui/screens/wallet/views.cljs]: @dmitryn Got it, thanks, will action! 👍

Comment (pacamara) [reply src/status_im/ui/screens/wallet/views.cljs]: @dmitryn Done :)  :muscle:

**Review by chadyj — COMMENTED** [review]
I tested with GBP, EUR, JPY, SEK, CLP and it looked good with correct decimals.

I tested with SEK and it showed the symbol like `kr352.07` but the convention is to show it after with a space like `352.07 kr`.   

Is there any easy fix to show symbol on the correct side?  If yes, would be great to see. If not, LGTM.

**Comment by pacamara** [comment]
@chadyj According to the goog.i18n source code it should be putting the kr at the end already, will try to figure out why that's not happening. :)

**Review by dmitryn — APPROVED** [review]

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/x9KUH5 for branch PR-4176

<blockquote><div><strong><a href="https://i.diawi.com/x9KUH5">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/CBsyLQ for branch PR-4176

<blockquote><div><strong><a href="https://i.diawi.com/CBsyLQ">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/fwWP5f for branch PR-4176

<blockquote><div><strong><a href="https://i.diawi.com/fwWP5f">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/hKeatt for branch PR-4176

<blockquote><div><strong><a href="https://i.diawi.com/hKeatt">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by pacamara** [comment]
@chadyj  @asemiankevich I've figured out the fix for the SEK symbol positioning problem and should be able to push it later today :)

**Comment by asemiankevich** [comment]
@pacamara great then i will test it later after your fix, thank you!

**Comment by pacamara** [comment]
@asemiankevich Have pushed the update now, it fixes the SEK problem, cheers!  :beers: 

(if a custom NumberSymbolsFormat_< locale > class is available corresponding to the currency code, set that as the default. Otherwise use NumberFormatSymbols_en. This is not necessary for correct currency symbol and decimal formatting, but is needed for correct positioning of the symbol).

**Comment by pacamara** [comment]
(repushed with indentation changes only)

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/kEimyM for branch PR-4176

<blockquote><div><strong><a href="https://i.diawi.com/kEimyM">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/rnN2CY for branch PR-4176

<blockquote><div><strong><a href="https://i.diawi.com/rnN2CY">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by asemiankevich** [comment]
@pacamara what serivce u are using to fetch currencies list? dont  see mine ( BYN , Belorussian Rouble). Also would be good to have search there in the list but it could be addressed in another issue

**Comment by pacamara** [comment]
@asemiankevich That was a previous bounty. :) The currencies list on the Profile -> Main Currency screen  was added by another contributor in #4067 to fix #4008. It's hardcoded in  `src/status_im/constants.cljs`

The `currency-code-to-nfs-map` added in this PR is grepped out of the `goog.i18n` source code at https://github.com/google/closure-library/blob/master/closure/goog/i18n/numberformatsymbols.js. It provides custom formatting of currencies, but it doesn't control the currencies list. :beers: 

<blockquote><div> GitHub</div><div><strong><a href="https://github.com/google/closure-library">google/closure-library</a></strong></div><div>closure-library - Google's common JavaScript library</div></blockquote>

**Comment by asemiankevich** [comment]
@pacamara sure, i mixed up the PRs in my head , thank you!

**Comment by pacamara** [comment]
@asemiankevich Welcome dude :cocktail:

**Pull request merged**